### PR TITLE
AI/cs-implementation

### PR DESCRIPTION
**Pull request by ayyubibrahimi**
- Share link points to sawt.eyeonsurveillance.org
- Integrated endless scroll
- Integrated seamless likes
- Integrated a function for reporting queries that are biased, inappropriate, hallucinations, etc
- Integrated seamless public comments

### CODE REVIEW

File: packages/web/components/CardResultsProvider.tsx [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
   cards: [],
   indexedCards: new Map<string, ICard>(),
+  hasMoreCards: true,
 });
 
 const compareCards = (a: ICard, b: ICard) => {
-  if (!a.created_at) return 1;
-  if (!b.created_at) return -1;
-  return new Date(b.created_at!).getTime() - new Date(a.created_at!).getTime();
+  if (!a.id) return 1;
```
Comment (marvinmarnold) [inline packages/web/components/CardResultsProvider.tsx:24]: Why this switch? Won't it cause cards not to be ordered chronologically?

File: packages/web/components/CardResultsProvider.tsx [hunk 2]
```diff
@@ -37,26 +62,50 @@ export default function CardResultsProvider({
...
   const [indexedCards, setIndexedCards] = useState(
     getIndexedCards(serverCards)
   );
+  const [hasMoreCards, setHasMoreCards] = useState(true);
+
+  const fetchMoreCards = async () => {
+    try {
+      const lastCard = cards[cards.length - 1];
+      const newCards = await fetchCardsFromSupabase(lastCard?.id);
+      console.log("Fetched new cards:", newCards);
```
Comment (marvinmarnold) [inline packages/web/components/CardResultsProvider.tsx:75]: Intentionally left?

File: packages/web/components/CardResultsProvider.tsx [hunk 3]
```diff
@@ -37,26 +62,50 @@ export default function CardResultsProvider({
...
+
+      if (newCards.length === 0) {
+        setHasMoreCards(false);
+      } else {
+        setCards((prevCards) => {
+          const updatedCards = [...prevCards, ...newCards];
+          console.log("Updated cards list:", updatedCards);
```
Comment (marvinmarnold) [inline packages/web/components/CardResultsProvider.tsx:82]: Also here.

File: packages/web/components/CardResultsProvider.tsx (hunk 2 above)
Comment (ayyubibrahimi) [reply packages/web/components/CardResultsProvider.tsx:99]: Don't follow. Same with below.

File: packages/web/components/CardResultsProvider.tsx (hunk 1 above)
Comment (ayyubibrahimi) [reply packages/web/components/CardResultsProvider.tsx]: Forgot to switch back. Just experimenting. Thinking about the ML model that will be responsible for sorting. What classification schema did you have in mind?

File: packages/web/components/CardResultsProvider.tsx (hunk 2 above)
Comment (marvinmarnold) [reply packages/web/components/CardResultsProvider.tsx:99]: Did you mean to leave the console.log statement? Usually good practice to remove non error logs before prod.

File: packages/web/components/CardResultsProvider.tsx (hunk 1 above)
Comment (marvinmarnold) [reply packages/web/components/CardResultsProvider.tsx]: I think sorting will always be chronological. When we have user accounts, we can base the filtering on personal preferences. Until then, I think showing all results is fine.

Then we'll need to mix in the case studies. Those may show up multiple times for the same user. The algorithm could probably be as simple as "one case study per rand(4, 8) cards"

**Pull request merged**